app/pulp/app/models/catalog.py

```diff
@@ -0,0 +1,48 @@
+"""
+Django models for artifact catalogs.
+"""
+from django.core import validators
+from django.db import models
+from django.utils.translation import ugettext as _
+
+from pulp.app.models import Model, Artifact, Importer
+
+
+class DownloadCatalog(Model):
+    """
+    A catalog that tracks where an :class:`Artifact` originated from
```

Comment:
Suggest something like:

Each catalog entry maps a URL  to an `Artifact` and to the `Importer` that contributed the entry.  The importer provides information needed to download the URL.

Since multiple importers can contribute catalog entries for the same artifact, the *originated from* is misleading.